src/facade/dragonfly_connection.cc

```diff
@@ -2105,7 +2138,7 @@ void Connection::EnsureMemoryBudget(unsigned tid) {
 
 Connection::WeakRef::WeakRef(std::shared_ptr<Connection> ptr, unsigned thread_id,
                              uint32_t client_id)
-    : ptr_{ptr}, thread_id_{thread_id}, client_id_{client_id} {
+    : ptr_{std::move(ptr)}, thread_id_{thread_id}, client_id_{client_id} {
```

Comment:
?

Reply:
deep copying a shared ptr increments the atomic counter so we pay for an atomic operation. move avoids that. I just saw the misuse and used move 😄